src-browser/worker_threads.js

```diff
@@ -18,31 +19,40 @@ export class Worker extends globalThis.Worker {
   };
 
   constructor(specifier, options) {
-    super(specifier, { ...(options || {}), type: 'module' });
-    EventEmitter.call(this);
-    this.addEventListener('error', (event) => this.emit('error', event.error || event.message));
-    this.addEventListener('messageerror', (event) => this.emit('messageerror', event.data));
-    this.addEventListener('message', (event) => this.emit('message', event.data));
-    this.postMessage({
+    super();
+    if (options?.eval === true) {
+      specifier = `data:text/javascript,${specifier}`;
+    }
+    const handle = this[kHandle] = new globalThis.Worker(specifier, {
+      ...(options || {}),
+      type: 'module',
+      // unstable
+      deno: { namespace: true },
+    });
```

Comment:
Avoid `globalThis` and object spread:

```suggestion
    const _global = typeof self !== 'undefined' ? self : typeof window !== 'undefined' ? window : typeof global !== 'undefined' ? global : null;
    const handle = this[kHandle] = Object.assign(new _global.Worker(specifier, Object.assign({}, options || {}, {
      type: 'module',
      // unstable
      deno: { namespace: true },
    }));
```

Reply:
Is there a reason for the `_global`?

Reply:
`globalThis` is a recent browser addition.